interop-testing/src/test/java/io/grpc/testing/integration/Http2NettyTest.java

```diff
@@ -68,16 +65,15 @@
   @Override
   protected NettyChannelBuilder createChannelBuilder() {
     try {
+      ChannelCredentials channelCreds = TlsChannelCredentials.newBuilder()
+          .keyManager(TestUtils.loadCert("client.pem"), TestUtils.loadCert("client.key"))
+          .trustManager(TestUtils.loadCert("ca.pem"))
+          .build();
       NettyChannelBuilder builder = NettyChannelBuilder
-          .forAddress(TestUtils.testServerAddress((InetSocketAddress) getListenAddress()))
+          .forAddress("localhost", ((InetSocketAddress) getListenAddress()).getPort(), channelCreds)
+          .overrideAuthority(TestUtils.TEST_SERVER_HOST)
```

Comment:
Why is this needed now? The setup hasn't changed - only using the new creds API AFAICS

Reply:
Take a [look at `TestUtils.testServerAddress()`](https://github.com/grpc/grpc-java/blob/3752b9e36590793ca1695faacabfe561a1b68305/testing/src/main/java/io/grpc/internal/testing/TestUtils.java#L70). It creates the address so the IP address points to localhost, but then injects TEST_SERVER_HOST as the "hostname" of the `InetSocketAddress`. That hostname is used as the default authority.

We currently don't have a `forAddress(SocketAddress, ChannelCreds)` API (which would be fair to add), but there's no good reason to continue using this approach. This code probably predates NameResolvers and `overrideAuthority()`. We actually don't want to use the `forAddress(SocketAddress)` often, as it disables the NameResolver.